README.md

```diff
@@ -1,12 +1,110 @@
+
+Alternatively, you can paste `jetbrains://gateway/com.coder.toolbox` into a browser.
+
+### Manual install
+
+There are two ways Coder Toolbox plugin can be installed. The first option is to manually download the plugin
+artifact from [JetBrains Marketplace](https://plugins.jetbrains.com/plugin/26968-coder/versions)
+or from [Coder's Github Release page](https://github.com/coder/coder-jetbrains-toolbox/releases).
+
+The next step is to copy the zip content to one of the following locations, depending on your OS:
+
+* Windows: `%LocalAppData%/JetBrains/Toolbox/cache/plugins/com.coder.toolbox`
+* macOS: `~/Library/Caches/JetBrains/Toolbox/plugins/com.coder.toolbox`
+* Linux: `~/.local/share/JetBrains/Toolbox/plugins/com.coder.toolbox`
+
+Alternatively, you can install it using the _Gradle_ tasks included in the project:
+
+```shell
+
+./gradlew cleanAll build copyPlugin
+```
+
+Make sure Toolbox is closed before running the command.
+
+## Connect to a Coder Workspace via JetBrains Toolbox URI
+
+You can use specially crafted JetBrains Gateway URIs to automatically:
+
+1. Open Toolbox
+
+2. Install the Coder Toolbox plugin (if not already installed)
+
+3. Connect to a specific Coder deployment using a URL and a token.
+
+4. Select a running workspace
+
+5. Install a specified JetBrains IDE on that Workspace
+
+6. Open a project folder directly in the remote IDE
+
+### Example URIs
+
+```text
+jetbrains://gateway/com.coder.toolbox?url=https%3A%2F%2Fdev.coder.com&token=zeoX4SbSpP-j2qGpajkdwxR9jBdcekXS2&workspace=bobiverse-bob&agent=dev&ide_product_code=GO&ide_build_number=241.23774.119&folder=%2Fhome%2Fcoder%2Fworkspace%2Fhello-world-rs
+
+jetbrains://gateway/com.coder.toolbox?url=https%3A%2F%2Fj5gj2r1so5nbi.pit-1.try.coder.app%2F&token=gqEirOoI1U-FfCQ6uj8iOLtybBIk99rr8&workspace=bobiverse-riker&agent=dev&ide_product_code=RR&ide_build_number=243.26053.17&folder=%2Fhome%2Fcoder%2Fworkspace%2Fhello-world-rs
+```
+
+### URI Breakdown
+
+```text
+jetbrains://gateway/com.coder.toolbox
+  ?url=http(s)://<your-coder-deployment>
+  &token=<auth-token>
+  &workspace=<workspace-name>
+  &agent/agent_id=<agent-name-or-agent-id>
+  &ide_product_code=<IDE-code>
+  &ide_build_number=<IDE-build>
+  &folder=<absolute-path-to-a-project-folder>
+```
+
+| Query param      | 	Description                                                                 | Mandatory |
+|------------------|------------------------------------------------------------------------------|-----------|
+| url              | 	Your Coder deployment URL (encoded)                                         | Yes       |
+| token            | 	Coder authentication token                                                  | Yes       |
+| workspace        | 	Name of the Coder workspace to connect to.                                  | Yes       |
+| agent            | 	Name of the agent associated with the workspace                             | No        |
```

Comment:
Let us remove the agent name from URI. It does not have any use. We already have the agent ID.

Reply:
There are multiple ways to get the ID programmatically, such as through API and within terrafomr. e.g. `coder_agent.<NAME>.id` gives the ID.